src/Microsoft.DotNet.CodeFormatting.Tests/AnalyzerFixerTestBase.cs

```diff
@@ -12,10 +12,31 @@ public abstract class AnalyzerFixerTestBase : CodeFormattingTestBase
     { 
         private IFormattingEngine _engine; 
 
-        protected AnalyzerFixerTestBase(IFormattingEngine engine)
-        { 
-            _engine = engine; 
-        } 
+        private IFormattingEngine Engine
+        {
+            get
+            {
+                if (_engine == null)
+                {
+                    _engine = FormattingEngine.Create();
+                }
+
+                return _engine;
+            }
+        }
+
+        protected void DisableAllDiagnostics()
```

Comment:
Methods to enable derived classes to enable only the one diagnostic they are testing. The rule-based test classes didn't need this mechanism because they worked by explicitly calling ProcessAsync on the single rule they were testing. But the analyzer-based tests work by calling (ultimately) CompilationWithAnalyzers.GetAnalyzerDiagnosticsAsync -- so I have to make sure that only the desired analyzer gets into the CompilationWithAnalyzers.